spec/lib/wipe_out/plugins/logger_spec.rb

```diff
@@ -1,59 +1,113 @@
+      expect(logged_messages).to match([
+        "[WipeOut] start plan=Plan(attributes=[:last_name, :access_tokens])",
+        "[WipeOut] executing plan=Plan(attributes=[:last_name, :access_tokens]) record_class=User id=#{user.id}",
+        "[WipeOut] executing plan=Plan(attributes=[]) record_class=Comment id=#{first_comment.id}",
+        "[WipeOut] wiped out plan=Plan(attributes=[]) record_class=Comment id=#{first_comment.id}",
+        "[WipeOut] executing plan=Plan(attributes=[]) record_class=Comment id=#{second_comment.id}",
+        "[WipeOut] wiped out plan=Plan(attributes=[]) record_class=Comment id=#{second_comment.id}",
+        "[WipeOut] wiped out plan=Plan(attributes=[:last_name, :access_tokens]) record_class=User id=#{user.id}",
+        "[WipeOut] completed plan=Plan(attributes=[:last_name, :access_tokens])"
+      ])
+    end
+
+    describe "nested root plans with plugins" do
+      it "uses plugin only from top level root plan that's executed" do
+        user_plan = WipeOut.build_plan do
+          plugin WipeOut::Plugins::Logger
+          wipe_out :last_name, :access_tokens
+          ignore :comments
+        end
+        user_basic_plan = WipeOut.build_plan do
+          include_plan(user_plan)
+          wipe_out :last_name
+          ignore :comments, :access_tokens
+        end
+        logger = double(Logger)
+        logged_messages = []
+        user_basic_plan.config.logger = logger
+        allow(logger).to receive(:debug) { |log| logged_messages << log }
+
+        plan.execute(create(:user))
+
+        expect(logged_messages).to match([])
+      end
+    end
   end
 
-  describe "nested root plans with plugins" do
-    it "uses plugin only from top level root plan that's executed" do
-      user_plan = WipeOut.build_plan do
+  context "when relation has multiple wipe out plans (plans union)" do
+    it "logs all executions for all records, even from relations", :aggregate_failures do
+      comments_plan = WipeOut.build_plan do
+        relation :resource_files do
+          on_execute { |execution| execution.record.destroy! }
+        end
+      end
+
+      vip_comments_plan = WipeOut.build_plan do
+        ignore :resource_files
+      end
+
+      plan = WipeOut.build_plan do
         plugin WipeOut::Plugins::Logger
+
         wipe_out :last_name, :access_tokens
-        ignore :comments
-      end
-      user_basic_plan = WipeOut.build_plan do
-        include_plan(user_plan)
-        wipe_out :last_name
-        ignore :comments, :access_tokens
+
+        relation :comments, plans: [comments_plan, vip_comments_plan] do |record|
+          record.value.starts_with?("[SPECIAL]") ? vip_comments_plan : comments_plan
+        end
       end
+
       logger = double(Logger)
       logged_messages = []
-      user_basic_plan.config.logger = logger
       allow(logger).to receive(:debug) { |log| logged_messages << log }
+      user = create(:user)
+      create_list(:comment, 2, user: user)
```

Comment:
I wanted to have 2 of each type of comment to make sure that for each plan with multiple records it will work fine.